src/bundle/DependencyInjection/EzPlatformRichTextExtension.php

```diff
@@ -177,4 +183,72 @@ private function validateCustomTemplatesConfig(
             }
         }
     }
+
+    /**
+     * Validate presence of inline Custom Tags in Toolbars.
+     *
+     * @param array $availableSiteAccesses a list of available SiteAccesses
+     * @param array $customTagsConfig Custom Tags configuration
+     * @param \Symfony\Component\DependencyInjection\ContainerBuilder $container
+     */
+    private function validateInlineCustomTagToolbarsConfig(
+        array $availableSiteAccesses,
+        array $customTagsConfig,
+        ContainerBuilder $container
+    ): void
+    {
+        $customTags = $this->getInlineCustomTags($customTagsConfig);
+        foreach ($this->getToolbarsBySiteAccess($availableSiteAccesses, $container) as $siteAccess => $toolbar) {
+            foreach ($toolbar as $toolbarName => $toolbarContent) {
+                // "text" toolbar is the only one that can contain inline tags
+                if ($toolbarName === 'text') {
+                    continue;
+                }
+
+                $buttons = $toolbarContent['buttons'];
+                foreach ($buttons as $buttonName => $buttonConfig) {
+                    if (in_array($buttonName, $customTags, true)) {
+                        throw new InvalidConfigurationException(
+                            sprintf(
+                                'Toolbar "%s" configured in "%s" cannot contain Custom Tag "%s". Inline Custom Tags are not allowed in Toolbars other than "%s".',
+                                $toolbarName,
+                                $siteAccess,
+                                $buttonName,
+                                'text'
+                            )
+                        );
+                    }
+                }
```

Comment:
CS/Readability: Could you extract some portion of this block to a separate method to reduce nesting level?

Reply:
Extracted the looping over toolbars to a separate method `checkForInlineTagsInToolbar`. I'm not sure if we want to further extract the button check part.